### PR TITLE
feat: expose OPL syntax check API

### PR DESCRIPTION
**Pull request by hperl**
We how have a new OPL API with a function to syntax-check an OPL file.

## Related issue(s)

Part of https://github.com/ory-corp/cloud/issues/2847

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/ory/graceful v0.1.3
 	github.com/ory/herodot v0.9.13
 	github.com/ory/jsonschema/v3 v3.0.7
-	github.com/ory/keto/proto v0.0.0-00010101000000-000000000000
+	github.com/ory/keto/proto v0.10.0-alpha.0
```
Comment (zepatrik) [inline go.mod:25]: Please revert, so there cannot be a mismatch.

File: internal/driver/config/provider.go [hunk 2]
```diff
@@ -155,23 +156,39 @@ func (k *Config) Set(key string, v any) error {
...
 }
 
-func (k *Config) WriteAPIListenOn() string {
+func (k *Config) MetricsListenOn() string {
+	return fmt.Sprintf(
+		"%s:%d",
+		k.p.String(KeyMetricsHost),
+		k.p.Int(KeyMetricsPort),
+	)
+}
```
Comment (zepatrik) [inline internal/driver/config/provider.go:178]: Maybe instead we could do

```suggestion
func (k *Config) listenOn(hostKey, portKey string) string {
	return fmt.Sprintf(
		"%s:%d",
		k.p.String(hostKey),
		k.p.Int(portKey),
	)
}

func (k *Config) MetricsListenOn() string {
	return k.listenOn(KeyMetricsHost, KeyMetricsPort)
}
```

**Review by zepatrik — CHANGES_REQUESTED** [review]
Nice :+1: just some minor naming suggestions, and we should be ready to :bus:

File: internal/driver/config/provider.go (hunk 2 above)
Comment (hperl) [reply internal/driver/config/provider.go:178]: Yes, I'll refactor. That part was nagging me as well :)

**Review by zepatrik — APPROVED** [review]

**Pull request merged**